x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/fields/create_field/create_field.tsx

```diff
@@ -124,59 +126,93 @@ export const CreateField = React.memo(function CreateFieldComponent({
     };
   };
 
-  const onTypeChange = (nextType: unknown) => {
-    const { subTypeOptions } = getSubTypeMeta(nextType as MainType);
-    form.setFieldValue('subType', subTypeOptions ? subTypeOptions[0].value : undefined);
+  const onTypeChange = (nextType: ComboBoxOption[]) => {
+    form.setFieldValue('type', nextType);
+
+    if (nextType.length) {
+      const { subTypeOptions } = getSubTypeMeta(nextType[0].value as MainType);
+      form.setFieldValue('subType', subTypeOptions ? [subTypeOptions[0]] : undefined);
+    }
   };
 
   const renderFormFields = useCallback(
-    ({ type }) => {
+    ({ type, subType }) => {
       const { subTypeOptions, subTypeLabel } = getSubTypeMeta(type);
 
+      const docLink =
+        (documentationService.getTypeDocLink(subType) as string) ||
```

Comment:
Range and Numeric are the only data types that still have sub-types, right? If so, then maybe we should just keep the help text under the "Type" combo box, since there are only Range and Numeric help pages, and no help pages for the individual sub-types.

https://www.elastic.co/guide/en/elasticsearch/reference/master/number.html
https://www.elastic.co/guide/en/elasticsearch/reference/master/range.html